temporal/t.rast.colors/t.rast.colors.py

```diff
@@ -172,19 +163,24 @@ def main():
         if equi:
             flags_ += "e"
 
-        try:
-            gs.run_command(
-                "r.colors",
-                flags=flags_,
-                file=filename,
-                color=color,
-                raster=raster,
-                volume=volume,
-                rules=rules,
-                overwrite=gs.overwrite(),
-            )
-        except CalledModuleError:
-            gs.fatal(_("Error in r.colors call"))
+        # Create the r.colors input file
+        with tempfile.NamedTemporaryFile(mode="w", delete=True) as temp_file:
+            temp_file.write("\n".join(str(row["id"]) for row in rows))
+            temp_file.flush()
+
+            try:
+                gs.run_command(
+                    "r.colors",
+                    flags=flags_,
+                    file=temp_file.name,
+                    color=color,
+                    raster=raster,
+                    volume=volume,
+                    rules=rules,
+                    overwrite=gs.overwrite(),
+                )
+            except CalledModuleError:
+                gs.fatal(_("Error in r.colors call"))
```

Comment:
Before, the file was closed but not deleted, and was reopened from r.colors.

Here, with delete=True, and just flushing, is it accessible correctly to the r.colors tool? Like would it complain that it can't read or write to the file as it is opened by another process? (It's a sentence we often hear on windows).

Do we know if a test runs this part of code?

Reply:
I suggest to go back to gs.tempfile and just focus on the original scope of the PR and apply the solution in t.rast.series below